### PR TITLE
{vis}[foss/2021a] MONAI v0.8.0 w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by crubb**
(created using `eb --new-pr`)

for @vacmar01.

Modelled after #12846.

Requires
* #14947
* #14949

### CODE REVIEW

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 10 out of 21 (2 easyconfigs in total)
easybuild-c7.novalocal - Linux UNKNOWN UNKNOWN, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/d8f13643698d851e667250c7d4464f31 for a full test report.

**Comment by verdurin** [comment]
I'm having trouble building this, though it relates to `PyTorch` - problems with the `git` checkout.

**Comment by crubb** [comment]
@verdurin - I noticed some `No space left on device` messages in the build log?

**Comment by verdurin** [comment]
Yes, I got past those - it now has a problem with the Git checkouts.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 6 out of 16 (2 easyconfigs in total)
centos7.lan - Linux UNKNOWN UNKNOWN, x86_64, Intel Core Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/ec0dc1cba1c7528677c94f94e60ff392 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 5 out of 10 (2 easyconfigs in total)
centos7.lan - Linux UNKNOWN UNKNOWN, x86_64, Intel Core Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/374ebcdb08cda1d841f55c4c5bdcb4bf for a full test report.

**Comment by crubb** [comment]
@verduin - Is there anything I can fix on my side? Thank you!

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
bear-pg0103u01a.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Gold 6330 CPU @ 2.00GHz (icelake), 2 x NVIDIA NVIDIA A100-PCIE-40GB, 470.57.02, Python 3.6.8
See https://gist.github.com/ec470c9a2a349e36e13e34a58e5859f7 for a full test report.

**Comment by branfosj** [comment]
I'm getting a different checksum:

```
$ sha256sum 0.8.0.tar.gz 
a63df7d5a680d9641c223ea090ff843a7d6f20bdb62095bd44f3b0480a4706ed  0.8.0.tar.gz
```

**Comment by crubb** [comment]
You're right. I still had the old archive, so here is the diff:

```
$ diff -ur MONAI-0.8.0-old/ MONAI-0.8.0-new/
diff -ur MONAI-0.8.0-old/monai/_version.py MONAI-0.8.0-new/monai/_version.py
--- MONAI-0.8.0-old/monai/_version.py	2021-11-25 18:16:28.000000000 +0000
+++ MONAI-0.8.0-new/monai/_version.py	2021-11-25 18:16:28.000000000 +0000
@@ -23,7 +23,7 @@
     # setup.py/versioneer.py will grep for the variable names, so they must
     # each be defined on a line of their own. _version.py will just call
     # get_keywords().
-    git_refnames = " (tag: 0.8.0, main)"
+    git_refnames = " (tag: 0.8.0)"
     git_full = "714d00dffe6653e21260160666c4c201ab66511b"
     git_date = "2021-11-25 18:16:28 +0000"
     keywords = {"refnames": git_refnames, "full": git_full, "date": git_date}
```

Apparently it happened when they released v0.8.1. I added the checksum.

File: easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+github_account = 'Project-MONAI'
+source_urls = [GITHUB_SOURCE]
+sources = ['%(version)s.tar.gz']
+checksums = [('b427c117250b24199ea88c0b3ec1a6d80a22958b4413b7c85c6318ef011f6479',
+              'a63df7d5a680d9641c223ea090ff843a7d6f20bdb62095bd44f3b0480a4706ed')]
+
+dependencies = [
+    ('CUDA', '11.3.1', '', True),
```
Comment (smoors) [inline easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a.eb:21]: why are you adding `CUDA` to the non-CUDA version? I would remove it unless really needed.

File: easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a.eb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+download_dep_fail = True
+
+preinstallopts = 'BUILD_MONAI=1'
+
+sanity_pip_check = True
+
+sanity_check_commands = ["python -c 'import monai; monai.config.print_config()'"]
+
+sanity_check_paths = {
+    'files': ['lib/python%(pyshortver)s/site-packages/%(namelower)s/_C.so'],
```
Comment (smoors) [inline easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a.eb:42]:
```suggestion
    'files': ['lib/python%%(pyshortver)s/site-packages/%%(namelower)s/_C.%s' % SHLIB_EXT],
```

File: easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a-CUDA-11.3.1.eb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+download_dep_fail = True
+
+preinstallopts = 'BUILD_MONAI=1'
+
+sanity_pip_check = True
+
+sanity_check_commands = ["python -c 'import monai; monai.config.print_config()'"]
+
+sanity_check_paths = {
+    'files': ['lib/python%(pyshortver)s/site-packages/%(namelower)s/_C.so'],
```
Comment (smoors) [inline easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a-CUDA-11.3.1.eb:43]:
```suggestion
    'files': ['lib/python%%(pyshortver)s/site-packages/%%(namelower)s/_C.%s' % SHLIB_EXT],
```

**Review by jfgrimm — COMMENTED** [review]
As decided in issue #16330, we have deprecated the use of `True` to signify a system-toolchain dependency (#16384), in favour of the more intuitive `SYSTEM` template constant. Due to the change in the test suite, please run `eb --sync-pr-with-develop 14951` and update the PR to use `SYSTEM` instead.

File: easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a.eb (hunk 1 above)
Comment (crubb) [reply easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a.eb]: That was an oversight - Thank you, consider it removed!

File: easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a-CUDA-11.3.1.eb (hunk 3 above)
Comment (crubb) [reply easybuild/easyconfigs/m/MONAI/MONAI-0.8.0-foss-2021a-CUDA-11.3.1.eb]: Done & Done. Thank you!

**Comment by crubb** [comment]
`SYSTEM` has been added to the `CUDA` dependency and I just `--sync-pr-with-develop`'ed.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @crubb!

**Pull request merged**